### PR TITLE
Adress workflow warnings

### LINKED ISSUE #14377: Github actions using deprecated set-output commands
## Expected Behavior

Workflows should complete without warnings.
See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/.

## Actual Behavior

The workflow shows warnings.

## Steps to Reproduce

Actions using old commands gives a warning like https://github.com/backstage/backstage/actions/runs/3228335759.

### PR DESCRIPTION
**Pull request by kevinneville**
Resolves #14377.

## Hey, I just made a Pull Request!

Improved GitHub workflows by formating files, bumping versions, and removing deprecates set-output commands.
See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! Are the double quotes a personal preference or is it an actual enhancement? We have single quotes pretty commonly set in our editors I think, so wanted to check if it really was a necessary change

**Comment by freben** [comment]
(Also, remember that you have to DCO sign all commits - the automated comment above has some more info)

**Comment by kevinneville** [comment]
> Hi! Are the double quotes a personal preference or is it an actual enhancement? We have single quotes pretty commonly set in our editors I think, so wanted to check if it really was a necessary change

Nope, I don't think it is needed, it was some automated lining and formating from my side. I dropped that commit now.

**Comment by kevinneville** [comment]
> (Also, remember that you have to DCO sign all commits - the automated comment above has some more info)

I think I have already done that. Or do I need to do something more.
```
Author: Kevin Neville <kevin@neville.se>
Date:   Fri Oct 28 17:21:49 2022 +0200

    remove deprecated set-output commands
    
    Signed-off-by: Kevin Neville <kevin@neville.se>
```

File: .github/workflows/pr-review-comment.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
             }
 
             const prNumber = artifact.name.slice('pr_number-'.length)
-            console.log(`::set-output name=pr-number::${prNumber}`);
+            console.log(`pr-number=${prNumber} >> $GITHUB_OUTPUT`);
```
Comment (Rugvip) [inline .github/workflows/pr-review-comment.yaml:36]: This doesn't work does it? I'm assuming we need to append to the file instead

File: scripts/check-if-release.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ async function main() {
 
   if (newVersions.length === 0) {
     console.log('No package version bumps detected, no release needed');
-    console.log(`::set-output name=needs_release::false`);
+    console.log(`needs_release=false >> $GITHUB_OUTPUT`);
```
Comment (Rugvip) [inline scripts/check-if-release.js:106]: Same thing with the rest of these

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thank you! 👍

**Pull request closed without merging**